CAP/gap/TerminalCategory.gi

```diff
@@ -63,106 +422,52 @@ end );
 ##
 InstallMethod( UniqueMorphism,
                [ IsCapCategory and IsTerminalCategory ],
-               
-  function( category )
-    local morphism, object;
+  
+  function( cat )
+    local object, morphism;
     
-    object := UniqueObject( CAP_INTERNAL_TERMINAL_CATEGORY );
+    object := UniqueObject( cat );
     
-    morphism := ObjectifyWithAttributes( rec( ), TheTypeOfCapTerminalCategoryMorphism,
-                                         Source, object,
-                                         Range, object,
-                                         IsOne, true );
+    morphism := rec( );
     
-    Add( CAP_INTERNAL_TERMINAL_CATEGORY, morphism );
+    ObjectifyMorphismWithSourceAndRangeForCAPWithAttributes( morphism, cat,
+                                           object,
+                                           object,
+                                           IsOne, true );
     
     SetIsWellDefined( morphism, true );
     
     return morphism;
     
 end );
 
-################################
-##
-## Category functions
-##
-################################
-
-##
-BindGlobal( "INSTALL_TERMINAL_CATEGORY_FUNCTIONS",
-            
-  function( )
-    local obj_function_list, obj_func, morphism_function_list, morphism_function, i;
-    
-    obj_function_list := [ AddZeroObject,
-                           AddKernelObject,
-                           AddCokernelObject,
-                           AddDirectProduct ];
-    
-    obj_func := function( arg ) return UniqueObject( CAP_INTERNAL_TERMINAL_CATEGORY ); end;
-    
-    for i in obj_function_list do
-        
-        i( CAP_INTERNAL_TERMINAL_CATEGORY, obj_func );
-        
-    od;
-    
-    morphism_function_list := [ AddIdentityMorphism,
-                                AddPreCompose,
-                                AddLiftAlongMonomorphism,
-                                AddColiftAlongEpimorphism,
-                                AddInverseForMorphisms,
-                                AddKernelEmbedding,
-                                AddKernelEmbeddingWithGivenKernelObject,
-                                AddKernelLiftWithGivenKernelObject,
-                                AddCokernelProjection,
-                                AddCokernelProjectionWithGivenCokernelObject,
-                                AddCokernelColift,
-                                AddCokernelColiftWithGivenCokernelObject,
-                                AddProjectionInFactorOfDirectProduct,
-                                AddProjectionInFactorOfDirectProductWithGivenDirectProduct,
-                                AddUniversalMorphismIntoDirectProduct,
-                                AddUniversalMorphismIntoDirectProductWithGivenDirectProduct ];
```

Comment:
Have you checked that all of those are also installed for the new version?

Reply:
Yes, they are installed.